File: frontend/src/game/upgrades/curses/health_reduc_3.ts
import { Upgrade, type UpgradeDef } from '../Upgrade'
import { UpgradeTargetID, UpgradeStatID, RarityID, UpgradeTypeID } from '../../data/ID'

export const HealthReduc3Def: UpgradeDef = {
  id: "health_reduc_3",
  name: "Reduced Health 3",
  description: "-20 max health.",
  rarity: RarityID.Rare,
  upgradeType: UpgradeTypeID.StatModifier,
  cost: 0,
  targetClass: UpgradeTargetID.Player,
  fieldInTargetClass: UpgradeStatID.MaxHealth,
  value: -20,
  isMultiplier: false,
  stackable: true,
  maxStacks: 99999,
  curse: true,
}

export class HealthReduc3 extends Upgrade {}

File: backend/app/core/upgrades/curses/health_reduc_3.py
from ..upgrade_implementation import UpgradeImplementation, UpgradeAppliers


class HealthReduc3(UpgradeImplementation):
    id = 'health_reduc_3'
    name = 'Reduced Health 3'
    description = '-20 max health.'
    rarity = 'rare'
    type = 'stat_modifier'
    cost = 0

    def can_apply(self, state: dict) -> bool:
        return True

    def apply(self, appliers: UpgradeAppliers) -> None:
        value = -20
        appliers.modifiers.add_multiplier('player', 'maxHealth', value)

    def validate(self, recorded_stats: dict) -> bool:
        return True
